python_modules/libraries/dagster-airflow/dagster_airflow/dagster_job_factory.py

```diff
@@ -249,3 +253,106 @@ def make_airflow_example_dags():
+        leaf_nodes = {
+            node_name.replace('airflow_', "") for node_name, downstream_nodes in forward_edges.items() if not downstream_nodes
+        }
+        asset_key_to_task_ids = {
+            dag.dag_id: leaf_nodes
+        }
+
+    output_mappings=set()
+    keys_by_output_name = {}
+    internal_asset_deps={}
+
+    visit_dict = {}
+    upstream_deps = set()
+
+    def find_upstream_dependency(solid_name: str) -> None:
+        # DFS to find all upsteam dependencies
+        # this node has been visited
+        if visit_dict[solid_name]:
+            return
+        # mark node as visted
+        visit_dict[solid_name] = True
+        # traverse upstream nodes
+        for output_handle in graph.dependency_structure.all_upstream_outputs_from_node(solid_name):
+            forward_node = output_handle.node_name
+            match = False
+            # find any assets produced by upstream nodes and add them to the internal asset deps
+            for asset_key in asset_key_to_task_ids:
+                if forward_node.replace('airflow_', "") in asset_key_to_task_ids[asset_key]:
+                    upstream_deps.add(AssetKey(asset_key))
+                    match = True
+            # don't traverse past nodes that have assets
+            if not match:
+                find_upstream_dependency(forward_node)
+
+    for asset_key in asset_key_to_task_ids:
+        asset_upstream_deps = set()
+        for task_id in asset_key_to_task_ids[asset_key]:
+            visit_dict = {s.name: False for s in graph.solids}
+            upstream_deps = set()
+            find_upstream_dependency(f"airflow_{task_id}")
+            for dep in upstream_deps:
+                asset_upstream_deps.add(dep)
+            internal_asset_deps[f"result_airflow_{task_id}"] = upstream_deps
+            keys_by_output_name[f"result_airflow_{task_id}"] = AssetKey(asset_key)
+            output_mappings.add(OutputMapping(
+                graph_output_name=f"result_airflow_{task_id}",
+                mapped_node_name=f"airflow_{task_id}",
+                mapped_node_output_name="airflow_task_complete" # Default output name
+            ))
+
+        # the tasks for a given asset should have the same internal deps
+        for task_id in asset_key_to_task_ids[asset_key]:
+            internal_asset_deps[f"result_airflow_{task_id}"] = asset_upstream_deps
+
+    new_graph = GraphDefinition(
+        name=graph.name,
+        node_defs=graph.node_defs,
+        dependencies=graph.dependencies,
+        output_mappings=list(output_mappings),
+    )
+
+    asset_def = AssetsDefinition.from_graph(
+        graph_def=new_graph,
+        partitions_def=TimeWindowPartitionsDefinition(
+            cron_schedule=cron_schedule,
+            timezone=dag.timezone.name,
+            start=start_date.strftime("%Y-%m-%d"),
+            fmt="%Y-%m-%d"
+        ) if isinstance(cron_schedule, str) and is_valid_cron_schedule(cron_schedule) else None,
```

Comment:
If it's not a valid cron schedule, I think we should probably avoid failing silently. Are there valid use cases for it not to be a valid cron schedule? Should we just error if it's not?